src/Store/ExtensionClientStoreConnector.re

```diff
@@ -455,6 +463,54 @@ let start = (extensions, setup: Setup.t) => {
       )
     });
 
+  let getOriginalUri = (bufferId, path, providers: list(SCM.Provider.t)) =>
+    Isolinear.Effect.createWithDispatch(~name="scm.getOriginalUri", dispatch => {
+      // Try our luck with every provider. If several returns Last-Writer-Wins
+      // TODO: Is there a better heuristic? Perhaps use rootUri to choose the "nearest" provider?
```

Comment:
TODO

Reply:
Thanks for calling this out! I think last-writer-wins is OK for now. Too bad there isn't a concept of 'precedence' with the SCM providers.